sjsonnet/src/sjsonnet/PrettyYamlRenderer.scala

```diff
@@ -67,26 +92,44 @@ class PrettyYamlRenderer(out: Writer = new java.io.StringWriter(),
     addSpaceAfterColon()
     flushBuffer()
     out.append(RenderUtils.renderDouble(d))
+    saveCurrentPos()
     out
   }
 
+  val loadedFileContents = mutable.Map.empty[Path, Array[Int]]
+  def saveCurrentPos() = {
+    val current = getCurrentPosition()
+    if (current != null){
+      bufferedComment = " # " + current.currentFile.renderOffsetStr(current.offset, loadedFileContents)
```

Comment:
This is in a hot code path (almost every line of output will have a comment), so maybe it would be better to avoid string concatenation. The logic where the buffer is appended to the `StringWriter` could do `out.append(" # ")` before appending the actual offset. Not sure if this would result in an observable speedup, but operations on Strings used to be a significant slowdown in the Scala compiler. Since this seems like an easy change, I'd be curious to know the outcome for very large files.

Reply:
I'll try this and see if there's any measurable difference